### PR TITLE
Randomize ports when running mds-policy-author and mds-policy-service tests

### PR DESCRIPTION
**Pull request by avatarneil**
## 📚 Purpose
This PR uses Jest's `setupFiles` config functionality in conjunction with `find-port-free-sync` to set random ports when running test suites which utilize multiple services, in order to avoid port collision.

## 📦 Impacts:
- mds-policy-author
- mds-policy-service

Note: because no package code was actually changed, just tests, no changeset was generated.

### CODE REVIEW

**Review by nessur — APPROVED** [review]
Sharp!!

File: packages/mds-policy-service/jest.config.js [hunk 1]
```diff
@@ -1 +1 @@
-module.exports = { ...require('../../jest.config.js') }
+module.exports = { ...require('../../jest.config.js'), setupFiles: ['./tests/configurePorts.ts'] }
```
Comment (drtyh2o) [inline packages/mds-policy-service/jest.config.js:1]: Any reason not to do this in the root config and just configure _all_ ports in one place? Would make it more obvious where and how to add new services.

File: .jest/configurePorts.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import findPort from 'find-port-free-sync'
+
```
Comment (drtyh2o) [inline .jest/configurePorts.ts:2]: Could use a helper function:
```ts
const randport = () => `${findPort()}`
```

File: .jest/configurePorts.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  process.env.INGEST_SERVICE_REPL_PORT = findPort().toString()
+
+  process.env.TRANSACTION_API_HTTP_PORT = findPort().toString()
+  process.env.TRANSACTION_SERVICE_REPL_PORT = findPort().toString()
+  process.env.TRANSACTION_SERVICE_RPC_PORT = findPort().toString()
+
+  console.log('Configured random ports!')
+}
+
+setPorts()
```
Comment (drtyh2o) [inline .jest/configurePorts.ts:49]: What, no IIFE? So unlike you. 😄

File: README.md [hunk 4]
```diff
@@ -293,3 +293,7 @@ curl --user user:password -X POST -F "jenkinsfile=<Jenkinsfile" http://localhost
 This assumes you have a jenkins server running on port 8080 of your local machine.  Note that this will only validate syntax, not whether the pipeline can actually be run.
 
 To commit code, you will need the pre-commit tool, which can be installed via `brew install pre-commit`.  For more information, see [SECURITY.md](.github/SECURITY.md)
+
+### Adding tests
+#### Using multiple services/APIs within one test suite
+In order to use multiple APIs/services within one test suite, it is important to ensure that random ports are assigned to each API/Service to avoid collision. To do this, we utilize Jest's setupFiles configuration in conjunction with a port randomization function. See [mds-policy-service/jest.config.js](packages/mds-policy-service/jest.config.js) and [mds-policy-service/tests/configurePorts.ts](packages/mds-policy-service/tests/configurePorts.ts) for an example of this pattern.
```
Comment (drtyh2o) [inline README.md:299]: This comment needs updating, I think.

File: env.jenkins [hunk 5]
```diff
@@ -1,23 +1,38 @@
 export AGENCY_API_HTTP_PORT=$(randport)
+
 export ATTACHMENT_SERVICE_REPL_PORT=$(randport)
 export ATTACHMENT_SERVICE_RPC_PORT=$(randport)
+
 export AUDIT_API_HTTP_PORT=$(randport)
+
+export AUDIT_SERVICE_REPL_PORT=$(randport)
```
Comment (drtyh2o) [inline env.jenkins:8]: Why is this file even needed still? Because not everything is using `jest` yet?

Comment (avatarneil) [reply env.jenkins:8]: Yep!

**Review by drtyh2o — APPROVED** [review]

**Pull request merged**